module/modmanager/modmaninterface/interface.go

```diff
@@ -0,0 +1,27 @@
+// Package modmaninterface abstracts the manager interface to avoid an import cycle/loop.
+package modmaninterface
+
+import (
+	"context"
+
+	"go.viam.com/rdk/config"
+	"go.viam.com/rdk/resource"
+)
+
+// ModuleManager abstracts the module manager interface.
+type ModuleManager interface {
+	AddModule(ctx context.Context, cfg config.Module) error
+
+	AddComponent(ctx context.Context, cfg config.Component, deps []string) (interface{}, error)
```

Comment:
Ideally, deps should/could be the registry.Dependencies here, so the conversion to strings is internal, but that causes an import cycle in lots of places, so taking the []string instead.